crates/notion-core/src/distro/package.rs

```diff
@@ -363,6 +370,87 @@ impl PackageVersion {
         }
         Ok(())
     }
+
+    /// Uninstall the specified package.
+    ///
+    /// This removes:
+    /// * the json config files
+    /// * the shims
+    /// * the unpacked and initialized package
+    ///
+    /// This optionally clears the inventory for this package, removing:
+    /// * the downloaded tarball and shasum files
+    pub fn uninstall(name: String, remove_all: bool) -> Fallible<()> {
+        // if the package config file exists, use that to remove any installed bins and shims
+        let package_config_file = path::user_package_config_file(&name)?;
+        if package_config_file.exists() {
+            let package_config = PackageConfig::from_file(&package_config_file)?;
+
+            for bin_name in package_config.bins {
+                let shim = path::shim_file(&bin_name)?;
+                fs::remove_file(shim).with_context(file_deletion_error)?;
+                let config_file = path::user_tool_bin_config(&bin_name)?;
+                fs::remove_file(config_file).with_context(file_deletion_error)?;
+                println!("Removed executable '{}' installed by '{}'", bin_name, name);
+            }
+
+            fs::remove_file(package_config_file).with_context(file_deletion_error)?;
+        } else {
+            // there is no package config - check for orphaned binaries
+            let user_bin_dir = path::user_bin_dir()?;
+            if user_bin_dir.exists() {
+                let orphaned_bins = binaries_from_package(&name)?;
+                for bin_name in orphaned_bins {
+                    let shim = path::shim_file(&bin_name)?;
```

Comment:
Do you think we can extract this into a separate function or rearrange the code to not repeat this block?

Reply:
Oh good call, I meant to extract that and forgot - I'll update that

Reply:
done!